### DISCUSSION TITLE
Should I avoid installing using pipx?

### DISCUSSION THREAD

**Discussion by thjbdvlt** [qutebrowser/qutebrowser#8312] [Q&A] [answered]
I am aware of the __virtual venv__ installation, but I am wondering if using `pipx` for that purpose should be avoided (because if i understand well what do `pipx` and `scripts/mkvenv.py`, it's kind of the same purpose)? Because I would like to use `pipx` (which i use for all other python globally available programs), and i see no mention of `pipx` in the documentation about installation.

**Comment by toofar** [comment, marked as answer]
If you just do `pipx install qutebrowser` I don't think Qt will be installed in the pipx venv (maybe we could support a `qutebrowser[qt]` optional target, but that isn't a thing currently).
If you have Qt installed by your distro `pipx --system-site-packages install qutebrowser` should work fine and end up with a very similar result to a basic mkvenv run.

mkvenv has a few more options and sanity checks, but if pipx works for you I see no reason to avoid it.

**Reply by The-Compiler** [reply]
I believe you can also do `pipx inject qutebrowser PyQt6 PyQt6-WebEngine` to install them after the environment is set up.

But yeah, `mkvenv.py` is mostly aimed at people who otherwise do not deal with Python stuff. I agree if you use pipx otherwise already, it makes sense to manage qutebrowser via that too.

**Reply by thjbdvlt** [reply]
> I believe you can also do `pipx inject qutebrowser PyQt6 PyQt6-WebEngine` to install them after the environment is set up.
> 
> But yeah, `mkvenv.py` is mostly aimed at people who otherwise do not deal with Python stuff. I agree if you use pipx otherwise already, it makes sense to manage qutebrowser via that too.

It completely works! This is great :-) Thank you.

**Comment by thjbdvlt** [comment]
Thank you, @toofar ! That's exactly what i wanted to know (that there is no security reason for the mkvenv install over a pipx install). As you say is your answer, Qt won't be installed, but that's no problem to me.
